File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int n = s.length(), cnt = 0;
        int i = n - 1;
        while(i >= 0 && !Character.isAlphabetic(s.charAt(i))) --i;
        while(i >= 0 && Character.isAlphabetic(s.charAt(i))) {
            --i;
            ++cnt;
        } 
        return cnt;
    }
}

File: 0058-length-of-last-word/0058-length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    function isAlphabet(str) {
        return /^[A-Za-z]+$/.test(str);
    }
    let cnt = 0;
    let n = s.length;
    let i = n - 1;
    while (i >= 0 && !isAlphabet(s[i])) {
        --i;
    }
    while (i >= 0 && isAlphabet(s[i--])) {
        ++cnt;
    }
    return cnt;
};